code/game/mecha/mecha.dm

```diff
@@ -205,6 +213,11 @@
 	GLOB.mechas_list -= src //global mech list
 	return ..()
 
+/obj/mecha/bullet_act(obj/item/projectile/P)
+	if (enclosed || !occupant || silicon_pilot)
+		return ..()
+	occupant.bullet_act(P) //If the sides are open, the occupant can be hit

```

Comment:
return value here ?

Reply:
This return stops the proc and makes the mech take damage as normally. It only passes the damage on to the pilot if one exists, as a non-silicon, and the mech has an open canopy.

Reply:
I mean return value from hitting the pilot needs to be passed back to projectile.

Reply:
Is that not handled by the occupant's bullet_act function?

Reply:
No, the caller won't get any information about penetration/blocking etc here. You need to return it.

Reply:
Alright, I _think_ I read how bullet_act is handled correctly, and have given a return BULLET_ACT_HIT to the function. I've also moved it to mecha_defense, where it should have been before anyway.